energy/energy_generation_consumption.py

```diff
@@ -4,76 +4,114 @@
 
 # COMMAND ----------
 
-# Source prevents programmatic data fetch: "https://www.energyinst.org/__data/assets/file/0015/1421601/2023-Country-Transition-Tracker-Consolidated-Format.csv"
-current_dir = Path().resolve()
-file_path = current_dir / '2023-Country-Transition-Tracker-Consolidated-Format.csv'
+# Source prevents programmatic data fetch: https://ember-climate.org/data/api/
```

Comment:
Not source that prevented programmatic data fetch, but rather corporate firewall settings. Let's update the comment to reflect it.